dwifslpreproc: Slice timing import numeric type

As reported on [forum](https://community.mrtrix.org/t/dwifslpreproc-invalid-slspec-file/2909).

Unlike `3.0_RC3`, which uses `explicit code` to import data from the specified "slspec" file, `dev` [uses](https://github.com/MRtrix3/mrtrix3/blob/95b144df3d9eb837fae08129cc292c3fb8490eac/bin/dwifslpreproc#L221) `matrix.load_numeric()` with `dtype=int` parameter. Presumably this is not working, as the observed "slspec" file in the scratch directory contains scientific notation floating-point numbers. So either the `dtype` parameter is not working as it should, or the function iss failing outright and instead the data are being read as [slice timing](https://github.com/MRtrix3/mrtrix3/blob/95b144df3d9eb837fae08129cc292c3fb8490eac/bin/dwifslpreproc#L225).

@ansk: Can you confirm whether or not [this warning](https://github.com/MRtrix3/mrtrix3/blob/95b144df3d9eb837fae08129cc292c3fb8490eac/bin/dwifslpreproc#L227) appears at the terminal? That would at least inform which of these two code paths is being used.

Hi, @Lestropie,

this warning does not appear. Here is my output of `dwifslpreproc`:

```
dwifslpreproc started at  Sun Oct 20 21:54:08 CEST 2019
dwifslpreproc: 
dwifslpreproc: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
dwifslpreproc: 
dwifslpreproc: Generated scratch directory: /vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/
Command:  mrconvert /vypoctysup3/skoch/THAL_dwi/data/THAL_C00615_20171209_1128_1/preproc/020_degibbsed.mif.gz /vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/dwi.mif -json_export /vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/dwi.json
dwifslpreproc: Changing to scratch directory (/vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/)
Command:  dirstat dwi.mif -output asym
Command:  mrconvert dwi.mif - | dwiextract - se_epi.mif -bzero
Command:  mrinfo dwi.mif -export_grad_mrtrix grad.b
dwifslpreproc: 1 spatial axis of DWIs has non-even size; this will be automatically padded for compatibility with topup, and the extra slice erased afterwards
Command:  mrconvert se_epi.mif -coord 2 56 - | mrcat se_epi.mif - se_epi_pad2.mif -axis 2
Command:  mrconvert dwi.mif -coord 2 56 - | mrcat dwi.mif - dwi_pad2.mif -axis 2
Command:  mrconvert se_epi_pad2.mif topup_in.nii -strides -1,+2,+3,+4 -export_pe_table topup_datain.txt
Command:  topup --imain=topup_in.nii --datain=topup_datain.txt --out=field --fout=field_map.nii.gz --config=/usr/local/fsl/etc/flirtsch/b02b0.cnf --verbose
Command:  mrinfo dwi_pad2.mif -export_pe_eddy applytopup_config.txt applytopup_indices.txt
Command:  dwiextract dwi_pad2.mif -pe 0.0,1.0,0.0,0.0347 - | mrconvert - dwi_pad2_pe_1.nii -json_export dwi_pad2_pe_1.json
Command:  applytopup --imain=dwi_pad2_pe_1.nii --datain=applytopup_config.txt --inindex=1 --topup=field --out=dwi_pad2_pe_1_applytopup.nii --method=jac
Command:  mrconvert dwi_pad2_pe_1_applytopup.nii.gz dwi_pad2_pe_1_applytopup.mif -json_import dwi_pad2_pe_1.json
Command:  dwiextract dwi_pad2.mif -pe 0.0,-1.0,0.0,0.0344 - | mrconvert - dwi_pad2_pe_2.nii -json_export dwi_pad2_pe_2.json
Command:  applytopup --imain=dwi_pad2_pe_2.nii --datain=applytopup_config.txt --inindex=2 --topup=field --out=dwi_pad2_pe_2_applytopup.nii --method=jac
Command:  mrconvert dwi_pad2_pe_2_applytopup.nii.gz dwi_pad2_pe_2_applytopup.mif -json_import dwi_pad2_pe_2.json
Command:  mrcat dwi_pad2_pe_1_applytopup.mif dwi_pad2_pe_2_applytopup.mif - -axis 3 | dwi2mask - - | maskfilter - dilate - | mrconvert - eddy_mask.nii -datatype float32 -strides -1,+2,+3
Command:  mrconvert dwi_pad2.mif eddy_in.nii -strides -1,+2,+3,+4 -export_grad_fsl bvecs bvals -export_pe_eddy eddy_config.txt eddy_indices.txt
Command:  eddy_cuda --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --topup=field --very_verbose --repol --cnr_maps --mporder=16 --s2v_niter=10 --slspec=slspec.txt --out=dwi_post_eddy --verbose
dwifslpreproc: CUDA version of 'eddy' was not successful; attempting OpenMP version
Command:  eddy_openmp --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --topup=field --very_verbose --repol --cnr_maps --mporder=16 --s2v_niter=10 --slspec=slspec.txt --out=dwi_post_eddy --verbose

dwifslpreproc: [ERROR] eddy_openmp --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --topup=field --very_verbose --repol --cnr_maps --mporder=16 --s2v_niter=10 --slspec=slspec.txt --out=dwi_post_eddy --verbose (dwifslpreproc:851)
dwifslpreproc: [ERROR] Information from failed command:
dwifslpreproc:
               EddyInputError:  Slice-to-vol is not yet implemented for the CPU version.
               Terminating program
dwifslpreproc:
dwifslpreproc: [ERROR] For debugging, inspect contents of scratch directory: /vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/
dwifslpreproc: Scratch directory retained; location: /vypoctysup3/skoch/THAL_dwi/data/dwifslpreproc-tmp-U0Y80W/
```

Antonin

--- a/bin/dwifslpreproc
+++ b/bin/dwifslpreproc
@@ -821,7 +821,7 @@ def execute(): #pylint: disable=unused-variable
       app.debug('New: ' + str(new_slice_groups))
       slice_groups = new_slice_groups
 
-    matrix.save_numeric('slspec.txt', slice_groups, header='')
+    matrix.save_numeric('slspec.txt', slice_groups, header='', fmt='%d')
     eddy_manual_options.append('--slspec=slspec.txt')
 
 

--- a/lib/mrtrix3/matrix.py
+++ b/lib/mrtrix3/matrix.py
@@ -156,7 +156,7 @@ def load_vector(filename, **kwargs): #pylint: disable=unused-variable
 
 # Save numeric data to a text file
 def save_numeric(filename, data, **kwargs):
-  fmt = kwargs.pop('fmt', '%.14e')
+  fmt = kwargs.pop('fmt', '%6f')
   delimiter = kwargs.pop('delimiter', ' ')
   newline = kwargs.pop('newline', '\n')
   add_to_command_history = bool(kwargs.pop('add_to_command_history', True))

